src/IotTelemetrySimulator/RunnerConfiguration.cs

```diff
@@ -188,36 +189,70 @@ public static RunnerConfiguration Load(IConfiguration configuration, ILogger log
                 }
             }
 
-            config.Intervals = LoadIntervals(configuration);
+            (config.Intervals, config.DeviceSpecificIntervals) = LoadIntervals(configuration);
             config.Header = GetTelemetryTemplate(configuration, nameof(Header), futureVariableNames);
             config.PartitionKey = GetTelemetryTemplate(configuration, nameof(PartitionKey), futureVariableNames);
 
             return config;
         }
 
-        private static Dictionary<string, int[]> LoadIntervals(IConfiguration configuration)
+        private static (int[], Dictionary<string, int[]>) LoadIntervals(IConfiguration configuration)
         {
-            var section = configuration.GetSection(nameof(RunnerConfiguration.Intervals));
+            var intervalForAllDevices = defaultInterval;
+            var section = configuration.GetSection(Constants.IntervalsConfigName);
             if (!section.Exists())
             {
-                return null;
+                section = configuration.GetSection(Constants.LegacyIntervalConfigName);
+                if (!section.Exists())
+                {
+                    return (intervalForAllDevices, null);
+                }
             }
 
             var result = new Dictionary<string, int[]>();
-
-            foreach (var intervalSection in section.GetChildren())
+            var sectionChildren = section.GetChildren();
+            if (sectionChildren.Any())
             {
-                if (intervalSection.Value != null && int.TryParse(intervalSection.Value, out int interval))
+                foreach (var intervalSection in sectionChildren)
                 {
-                    result[intervalSection.Key] = new[] { interval };
+                    int[] interval = null;
+                    if (intervalSection.Value != null && int.TryParse(intervalSection.Value, out int singleInterval))
+                    {
+                        interval = new[] { singleInterval };
+                    }
+                    else
+                    {
+                        interval = intervalSection.Get<int[]>();
+                    }
+
+                    if (string.IsNullOrEmpty(intervalSection.Key))
+                    {
+                        intervalForAllDevices = interval;
+                    }
+                    else
+                    {
+                        result[intervalSection.Key] = interval;
+                    }
                 }
-                else
+            }
+            else if (!string.IsNullOrEmpty(section.Value))
+            {
+                var settingsInterval = section.Value.Split(new[] { ",", ";" }, StringSplitOptions.RemoveEmptyEntries)
+                    .Select(x =>
+                    {
+                        int.TryParse(x, out var intX);
+                        return intX;
+                    })
+                    .Where(x => x > 0)
```

Comment:
We validate this in only one if branch. Perhaps do this before `return`, and throw and exception if there are non-positives?

Reply:
Good point. We are not raising an error in case the value is not a positive number.